lib/AST/ASTMangler.cpp

```diff
@@ -1733,7 +1734,7 @@ void ASTMangler::appendProtocolConformance(const ProtocolConformance *conformanc
     appendIdentifier(
               fileUnit->getDiscriminatorForPrivateValue(behaviorStorage).str());
     appendProtocolName(conformance->getProtocol());
-    appendIdentifier(behaviorStorage->getName().str());
+    appendIdentifier(behaviorStorage->getBaseName().getBaseName().str());
```

Comment:
This basically asserts that the base name is not special because then the second `getBaseName()` will crash. Maybe an `assert` would indeed be cleaner.